app/src/forms/auth/middleware/userAccess.js

```diff
@@ -133,29 +132,41 @@ const hasFormPermissions = (permissions) => {
   };
 };
 
+/**
+ * Express middleware to check that the caller has the given permissions for the
+ * submission identified by params.formSubmissionId or query.formSubmissionId.
+ * This will fall through if everything is OK, otherwise it will call next()
+ * with a Problem describing the error.
+ *
+ * @param {string[]} permissions the access the user needs for the submission
+ * @returns nothing
+ */
 const hasSubmissionPermissions = (permissions) => {
   return async (req, _res, next) => {
     try {
-      // Skip permission checks if requesting as API entity
+      // Skip permission checks if req is already authorized using an API key.
       if (req.apiUser) {
-        return next();
+        next();
+
+        return;
```

Comment:
Just doing this for readability - next doesn't actually return anything, so the `return next()` shortcut can be a bit confusing at first.